assets/src/edit-story/utils/blobs.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Blob cache.
+ *
+ * @type {Object.<string, File>}
+ */
+const blobCache = {};
+
+/**
+ * Create a blob URL from a file.
+ *
+ * @param {File} file File object.
+ * @return {string} Blob URL.
+ */
+export function createBlob(file) {
+  const url = window.URL.createObjectURL(file);
+
+  blobCache[url] = file;
```

Comment:
So this is a cache for blob used before? We are just using a variable as a cache. Is there any risk this? Wouldn't some state make sense here?

Reply:
No risk really. This variable is a simple key value store that doesn't need something fancy.

Reply:
Is there chance the browser clear the variable, maybe when the browser does it's garbage collection to clear memory? We are talking about a browser window with the editor maybe open for 30+ minutes.

Reply:
As long as a variable is still being used a browser would not simply discard it.

Reply:
Also note that every blob is removed again from this cache once upload is finished. So they're not in this array forever.